### ISSUE TITLE
Incorrect scope check for let declaration

### ISSUE THREAD

**Issue by pasha-mf** [facebook/flow#895]
```
/* @flow */
while (true) {
  let x = Math.random();
  if (x > 0.5) break; /* works fine */
  if (x) /* somehow this produces an error */;
}

let x = 2;
```

**Comment by samwgoldman** [comment]
Yeah, that's wrong. I thought it was maybe related to the `break`, but this errors too:

```javascript
/* @flow */
while (true) {
  let x;
  x = 0;
}
let x = 0;
```
```
test.js:2:1,5:1: x
let referenced before declaration
test.js:6:5,5: let x
```

**Comment by samwgoldman** [comment]
OK, I think I figured this out. The ridiculous error position (pointing to the `while` statement itself` was a dead giveaway. Working on a fix now.

**Comment by samwgoldman** [comment]
The inimitable @bhosmer is cooking up a fix for this one. Probably not for 0.17, but soon.

**Comment by samwgoldman** [comment]
Fixed in 0.18 :)
